### DISCUSSION TITLE
Why is this template deleting the previous line?

### DISCUSSION THREAD

**Discussion by danielo515** [SilentVoid13/Templater#1184] [Help]
Let's say I have this template:

```
---
<%* 
const formApi = app.plugins.plugins.obsidianModalForm.api;
const result = await formApi.exampleForm();
tR = result.asFrontmatterString();
-%>
---
```

As you can sii, I'm directly assigning a result to the tR value.
That, when inserted produces an output that contains my string, the final `---` but mimsses the preceding `---`. So it ends up like
```
result text
---
```
If instead I append, like this:

```
---
<%* 
const formApi = app.plugins.plugins.obsidianModalForm.api;
const result = await formApi.exampleForm();
tR += result.asFrontmatterString();
-%>
---
```

The result is the expected. Why is this?

**Comment by Zachatoo** [comment]
That is the [documented](https://silentvoid13.github.io/Templater/commands/execution-command.html#how-to-output-a-value-from-a-javascript-execution-command-) way to use `tR`. You should always use `+=` unless you want to override the file contents up until that point in the template.

**Reply by Zachatoo** [reply]
I use the word documented, the documentation isn't clear on what happens if you use `=`, but all the examples use `+=`.
